### PR TITLE
Remove showdupesfromrepos config option

### PR DESCRIPTION
**Pull request by m-blaha**
- the option is not documented even for dnf4
- the name is kind of cryptic

For commands that use it (list, info, search), the functionality is replaced by --showduplicates command line option.

### CODE REVIEW

File: dnf5/commands/list/list.cpp [hunk 1]
```diff
@@ -58,12 +57,8 @@ void ListCommand::set_argument_parser() {
...
     cmd.register_positional_arg(specs);
 
-    auto showduplicates = parser.add_new_named_arg("showduplicates");
-    showduplicates->set_long_name("showduplicates");
-    showduplicates->set_description("Show all versions of the packages, not only the latest ones.");
-    showduplicates->set_const_value("true");
-    showduplicates->link_value(&config.get_showdupesfromrepos_option());
-    cmd.register_named_arg(showduplicates);
+    show_duplicates = std::make_unique<libdnf::cli::session::BoolOption>(
+        *this, "showduplicates", '\n', "Show all versions of the packages, not only the latest ones.", false);
```
Comment (jan-kolarik) [inline dnf5/commands/list/list.cpp:61]: I am just wondering, why is the short name for the command == '\n'. Does it have some special hidden meaning?

Comment (m-blaha) [reply dnf5/commands/list/list.cpp:61]: To be honest, I've copy/pasted it from somewhere. But good question, I'll investigate.

Comment (m-blaha) [reply dnf5/commands/list/list.cpp:61]: Well, it should be `\0`

Comment (m-blaha) [reply dnf5/commands/list/list.cpp]: So it looks like it wasn't copy/paste, but my invention. Thanks for spotting this!. Fixed.

**Review by jan-kolarik — APPROVED** [review]
LGTM, thanks!

**Pull request merged**